node/graphql/index.ts

```diff
@@ -74,14 +75,30 @@ export const resolvers = {
   Query: {
     getStores: async (_: any, param: any, ctx: any) => {
       const {
-        clients: { hub, sitemap },
+        clients: { hub, sitemap, vbase },
         vtex: { logger },
       } = ctx
 
       try {
         sitemap.hasSitemap().then((has: any) => {
           if (has === false) {
-            sitemap.saveIndex()
+            const APP_NAME = 'store-locator'
+            const SHCEMA_NAME = 'sitemap'
+
+            vbase
+              .getJSON(APP_NAME, SHCEMA_NAME, true)
+              .then((getResponse: any) => {
+                const { alreadyHasSitemap } = getResponse
+
+                !alreadyHasSitemap &&
+                  sitemap.saveIndex().then((res: any) => {
+                    if (res.data.saveIndex) {
+                      vbase.saveJSON(APP_NAME, SHCEMA_NAME, {
+                        alreadyHasSitemap: true,
+                      })
+                    }
+                  })
+              })
```

Comment:
You must use the try/catch with await/async, in this case, if vbase triggers some error this code won't catch that.

Reply:
Agreed @arturmagalhaesjr , i changed this in order to use a catch for both calls, get vbase and save in vbase.